### PR TITLE
Correctly use limits middleware in series query

### PR DESCRIPTION
**Pull request by chaudum**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

**Comment by cyriltovena** [comment]
Yep that seems good

File: pkg/querier/queryrange/roundtrip.go [hunk 1]
```diff
@@ -334,7 +334,7 @@ func NewLabelsTripperware(
 	retryMiddlewareMetrics *queryrange.RetryMiddlewareMetrics,
 	splitByMetrics *SplitByMetrics,
 ) (queryrange.Tripperware, error) {
-	queryRangeMiddleware := []queryrange.Middleware{StatsCollectorMiddleware(), NewLimitsMiddleware(limits)}
```
Comment (DylanGuedes) [inline pkg/querier/queryrange/roundtrip.go:337]: double checking: here the `StatsCollectorMiddleware()` is removed. Doesn't that cause us to lose the statistics? :thinking:

Comment (chaudum) [reply pkg/querier/queryrange/roundtrip.go:337]: The labels query does not return any stats, and the unintended added middleware would only log a warning:
https://github.com/grafana/loki/blob/main/pkg/querier/queryrange/stats.go#L100

Comment (DylanGuedes) [reply pkg/querier/queryrange/roundtrip.go:337]: got it, ty!

**Review by cyriltovena — APPROVED** [review]
LGTM thanks !

**Pull request merged**